src/object-store/src/manager.rs

```diff
@@ -0,0 +1,112 @@
+// Copyright 2023 Greptime Team
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::collections::HashMap;
+
+use snafu::ensure;
+
+use crate::error::Result;
+use crate::{error, ObjectStore};
+
+/// Manages multiple object stores so that users can configure a storage for each table.
+/// This struct certainly have one default object store, and can have zero or more custom object stores.
+pub struct ObjectStoreManager {
+    stores: HashMap<String, ObjectStore>,
+    default_object_store: ObjectStore,
+}
+
+impl ObjectStoreManager {
+    pub fn try_new(
```

Comment:
nitpick: Document that `stores` must contain `default_object_store`.
```suggestion
    /// Creates a new manager with specific object stores. Returns an error if `stores` doesn't contain the default object store.
    pub fn try_new(
```

Reply:
I addressed your review: https://github.com/GreptimeTeam/greptimedb/commit/09fc1d7d45b68c1f03e0cfe4839bebdc000e6b8e